### PR TITLE
FragPipe+MsFragger argument parsing (+fixing Proline Parsing Errors)

### PR DESCRIPTION
**Pull request by enryH**
- fixed_mods, variable_mods have a different format
- fdr: given as integers, transformed to prop. (1 -> 0.01)

update FragPipe/MSFragger run to latest provided by Klemens

Will also address other issues with parsing parameters currently discussed.

### CODE REVIEW

**Review by RobbinBouwmeester — APPROVED** [review]

**Comment by enryH** [comment]
I think we still need to update the UI as FragPipe now has two parameter files.

**Comment by wolski** [comment]
@enryH @RobbinBouwmeester 
Do we need two different test frameworks
import unittest in test_module_dda_quant
and 
import pytest
in test_parse_params_proline.py
?

**Comment by wolski** [comment]
@enryH @RobbinBouwmeester 

# ModuleNotFoundError: No module named 'pytest'

I have a problem with the pytest within the ProteoBench evniroment:
I did set up the env with 
pip install flit
flit install -s


```
Failed to import test module: test_parse_params_proline
Traceback (most recent call last):
  File "/Users/witoldwolski/anaconda3/envs/ProteoBench/lib/python3.10/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/Users/witoldwolski/anaconda3/envs/ProteoBench/lib/python3.10/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/Users/witoldwolski/__projects/ProteoBench/test/test_parse_params_proline.py", line 5, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'
```

Any idea what might be wrong?

**Comment by wolski** [comment]
Adding "pytest" to the dependencies in the pyproject.toml fixes the import issue.

However, the pytest tests are not shown in my VS code 

image

, and running pytest gives

image

**Comment by enryH** [comment]
Flit installs the dependencies for the package, not the development dependencies. So you will need to install `pytest` manually in your local environment.

File: proteobench/io/params/fragger.py [hunk 1]
```diff
@@ -46,13 +54,85 @@ def read_file(file: str) -> list[Parameter]:
...
+    params.software_name = "FragPipe"
+    params.software_version = header
+    params.search_engine = "MSFragger"
+
+    msfragger_executaable = fragpipe_params.loc["fragpipe-config.bin-msfragger", "value"]
+    msfragger_executaable = Path(msfragger_executaable).name
+    match = re.search(VERSION_NO_PATTERN, msfragger_executaable)
+
+    if match:
+        msfragger_executaable = match.group()
```
Comment (mlocardpaulet) [inline proteobench/io/params/fragger.py:86]: is the double "a" in purpose?

**Review by wolski — APPROVED** [review]
All good. The problem with the test seems to be related to my system. I will investigate it further.

**Comment by mlocardpaulet** [comment]
Also, when I run it locally it does not print the points in https://github.com/Proteobench/Results_Module2_quant_DDA/ anymore. I don't know if the issue is from me.

**Review by mlocardpaulet — APPROVED** [review]
looks good to me.

**Comment by mlocardpaulet** [comment]
We still need to work on the FragPipe parameter parsing side. We need to upload two files, so need to modify the input to allow this.

**Comment by brvpuyve** [comment]
Hi all,

Sorry for the delay. 
As far as I understood from Henry, we were planning to use the fragpipe.workflow file. 
From this file, most of the parameters can be extracted. 
Example:
MSFragger version - Line 53 fragpipe-config.bin-msfragger=C\:\\Users\\brvpuyve\\Downloads\\FragPipe-21.1\\fragpipe\\tools\\MSFragger-4.0\\MSFragger-4.0.jar
Database being used - Line 6 database.db-path=Q\:\\MISC_PERSONAL\\Bart\\Proteobench\\2024-01-08-decoys-BenchmarkFASTAModule2_DDA.fasta.fas
Match Between runs - Line 66 ionquant.mbr=1 (1 being activated)
Precursor Mass Tolerance - Line 148 msfragger.precursor_true_tolerance=20
Fragment Mass Tolerance - Line 107 msfragger.fragment_mass_tolerance=0.02
Fixed modifications - Line 165 msfragger.table.fix-mods=0.0,C-Term Peptide,true,-1; 0.0,N-Term Peptide,true,-1; 0.0,C-Term Protein,true,-1; 0.0,N-Term Protein,true,-1; 0.0,G (glycine),true,-1; 0.0,A (alanine),true,-1; 0.0,S (serine),true,-1; 0.0,P (proline),true,-1; 0.0,V (valine),true,-1; 0.0,T (threonine),true,-1; 57.02146,C (cysteine),true,-1; 0.0,L (leucine),true,-1; 0.0,I (isoleucine),true,-1; 0.0,N (asparagine),true,-1; 0.0,D (aspartic acid),true,-1; 0.0,Q (glutamine),true,-1; 0.0,K (lysine),true,-1; 0.0,E (glutamic acid),true,-1; 0.0,M (methionine),true,-1; 0.0,H (histidine),true,-1; 0.0,F (phenylalanine),true,-1; 0.0,R (arginine),true,-1; 0.0,Y (tyrosine),true,-1; 0.0,W (tryptophan),true,-1; 0.0,B ,true,-1; 0.0,J,true,-1; 0.0,O,true,-1; 0.0,U,true,-1; 0.0,X,true,-1; 0.0,Z,true,-1
Variabel modifications - Line 166 msfragger.table.var-mods=15.9949,M,true


Best,
Bart

**Comment by enryH** [comment]
It's done already, but we need the `.workflow` and the `.params` file  - so it's two

**Pull request merged**